kategory/src/main/kotlin/kategory/typeclasses/Composed.kt

```diff
@@ -0,0 +1,121 @@
+package kategory
+
+/**
+ * https://www.youtube.com/watch?v=wvSP5qYiz4Y
+ */
+interface ComposedType<out F, out G>
+
+@Suppress("UNCHECKED_CAST")
+fun <F, G, A> HK<F, HK<G, A>>.lift(): HK<ComposedType<F, G>, A> = this as HK<ComposedType<F, G>, A>
+
+@Suppress("UNCHECKED_CAST")
+fun <F, G, A> HK<ComposedType<F, G>, A>.lower(): HK<F, HK<G, A>> = this as HK<F, HK<G, A>>
+
+interface ComposedFoldable<in F, in G> :
+        Foldable<ComposedType<F, G>> {
+
+    fun FF(): Foldable<F>
+
+    fun GF(): Foldable<G>
+
+    override fun <A, B> foldL(fa: HK<ComposedType<F, G>, A>, b: B, f: (B, A) -> B): B =
+            FF().foldL(fa.lower(), b, { bb, aa -> GF().foldL(aa, bb, f) })
+
+    fun <A, B> foldLC(fa: HK<F, HK<G, A>>, b: B, f: (B, A) -> B): B = foldL(fa.lift(), b, f)
```

Comment:
Since https://github.com/kategory/kategory/pull/172 detekt enforces staying on the same line